### ISSUE TITLE
Link to Owned Music in the Discography

### ISSUE THREAD

**Issue by ReiDuran** [Bambosh/unofficial-homestuck-collection#22]
The discography is set up great, but it would be kind of cool if you could link to music that you already own instead of needing to stream it from Bandcamp. Obviously it's good to link to the music in order to support the artists, but being able to link albums you've previously already purchased would help the Collection be as offline as possible.

Honestly, this is a minor little nitpick in the grand scheme of things. The Collection is crazy extensive and honestly this is the only real "improvement" that I can even think of.

**Comment by ShadowTheEdgelord** [comment]
I second this, perhaps maybe just set up a little embedded player that points to the offline directory and we supply our own owned song files into correct folders

**Comment by Bambosh** [comment]
Not super likely, due to the infinite different ways people their albums stored. Original releases or bundled together, mp3 or FLAC, custom filenames, the goes on. The only way to accomplish this would be to make people either enter out the urls for every track by hand, or to arrange their own tracks into a specific folder+filename structure, which would both be maddening.

**Comment by GiovanH** [comment]
I do like the idea of setting up the spec to allow songs to have local file paths, and configuring the client to play files from disc, even if the client doesn't directly match files to tracks itself. That mapping is something mods could do.

**Comment by ReiDuran** [comment]
> Not super likely, due to the infinite different ways people have their albums stored. Original releases/bundled together, mp3/FLAC, custom filenames, the goes on. The only way to accomplish this would be to make people either enter out the urls for every track by hand, or to arrange their own tracks into a specific folder+filename structure, which would both be maddening.

Yeah, I guess that does make sense. It'd kind of be a big issue to link everything manually.

**Comment by Stalzak** [comment]
i've been thinking about this topic for a while and even tho i'm not really sure about how doable/safe it would be- maybe a recognition of music files by their hashes? that way the user only needs to assign a folder in which each file could be hashed, and if a match's found it would then be easy to set up the pathname locating the file for the given music.

it would require calculating the hashes of each and every music in the discography to know their respective expected value, but it would have a lots of pros on the long run- no manual set-up on a file per file basis, no problem caused by file name changes, no issues on missing songs or albums... would love to hear your thoughts on that idea!

**Comment by Sixthhokage1** [comment]
There are 603 tracks in Homestuck's discography, according to the Wikipedia article on the "Music of Homestuck". Bandcamp offers downloads in 8 formats (MP3 320, MP3 V0, FLAC, AAC, Ogg Vorbis, ALAC, WAV, AIFF). That's 4824 hashes needed to fully support the official downloads from Bandcamp.

**Comment by Stalzak** [comment]
welp, that'd be a massive commitment for sure, but nothing some command line couldn't handle. altho, i did forget about metadata and how some users could have altered them in their collection- apparently some tools exist to compute only the music bites without metadata, but i'm not sure if that kind of tool exists for every file format.
